speechbrain/dataio/sampler.py

```diff
@@ -578,6 +585,15 @@ def _generate_batches(self):
             item_len = self._ex_lengths[str(idx)]
             # bucket to fill up most padding
             bucket_id = np.searchsorted(self._bucket_boundaries, item_len)
+
+            if (
+                len(bucket_batches[bucket_id])
```

Comment:
This change is a bit more mysterious to me

Reply:
This is this code: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/dataio/sampler.py#L595  that has been moved here following Andreas PR.